libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -180,17 +180,22 @@ public static synchronized void reset() {
      * @param account User account.
      */
     public static synchronized void reset(UserAccount account) {
-        Set<String> keysToRemove = new HashSet<>();
-        for (String key : INSTANCES.keySet()) {
-            if (key.startsWith(account.getUserId())) {
-                keysToRemove.add(key);
-                SyncManager syncManager = INSTANCES.get(key);
-                syncManager.threadPool.shutdownNow();
+        if (account == null) {
```

Comment:
@narakim I don't think we should do this. This API specifically expects a valid `account` instance.  Why are we substituting it with `currentUser` if an `account` isn't passed in? I'm okay with the `null` check below though.

Reply:
Will do! Thank you Bharath!